diff -u -r arb-2.8.1.orig/configure arb-2.8.1/configure
--- arb-2.8.1.orig/configure	2015-12-31 17:30:01.000000000 +0100
+++ arb-2.8.1/configure	2016-01-20 16:41:41.336726596 +0100
@@ -647,6 +647,7 @@
 echo "ARB_SHARED=$SHARED" >> Makefile
 echo "ARB_LIB=$ARB_LIB" >> Makefile
 echo "ARB_LIBNAME=$ARB_LIBNAME" >> Makefile
+echo "ARB_MAJOR=$ARB_MAJOR" >> Makefile
 echo "ARB_SOLIB=$ARB_SOLIB" >> Makefile
 echo "EXEEXT=$EXEEXT" >> Makefile
 echo "PREFIX=$PREFIX" >> Makefile
diff -u -r arb-2.8.1.orig/Makefile.in arb-2.8.1/Makefile.in
--- arb-2.8.1.orig/Makefile.in	2015-12-31 17:30:01.000000000 +0100
+++ arb-2.8.1/Makefile.in	2016-01-20 16:30:32.575298517 +0100
@@ -101,6 +101,7 @@
 		$(LDCONFIG) -n "$(CURDIR)"; \
 	fi
 	ln -sf "$(ARB_LIB)" "$(ARB_LIBNAME)"; \
+	ln -sf "$(ARB_LIB)" "$(ARB_LIBNAME).$(ARB_MAJOR)"; \
 
 libarb.a: $(OBJS) $(LIB_SOURCES) $(EXT_SOURCES) $(HEADERS) $(EXT_HEADERS) | build build/interfaces
 	$(AT)$(foreach ext, $(EXTENSIONS), $(foreach dir, $(patsubst $(ext)/%.h, %, $(wildcard $(ext)/*.h)), mkdir -p build/$(dir); BUILD_DIR=$(CURDIR)/build/$(dir); export BUILD_DIR; MOD_DIR=$(dir); export MOD_DIR; $(MAKE) -f $(CURDIR)/Makefile.subdirs -C $(ext)/$(dir) static || exit $$?;))
