From 71ff9e769ba5d9995b367201f0d41b7a8dedab9d Mon Sep 17 00:00:00 2001
From: John Bowman <bowman@ualberta.ca>
Date: Sat, 14 Nov 2015 01:25:56 -0700
Subject: [PATCH] Support GSL 2.0.

---
 gsl.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gsl.cc b/gsl.cc
index b500557..0f81dc6 100644
--- a/gsl.cc
+++ b/gsl.cc
@@ -27,6 +27,7 @@
 #include <gsl/gsl_rng.h>
 #include <gsl/gsl_randist.h>
 #include <gsl/gsl_cdf.h>
+#include <gsl/gsl_version.h>
 
 #include "opsymbols.h"
 
@@ -1088,7 +1089,11 @@ void gen_rungsl_venv(venv &ve)
   addGSLDOUBLE2Func<gsl_sf_ellint_F>(SYM(F));
   addGSLDOUBLE2Func<gsl_sf_ellint_E>(SYM(E));
   addGSLDOUBLE3Func<gsl_sf_ellint_P>(SYM(P),SYM(phi),SYM(k),SYM(n));
+#if GSL_MAJOR_VERSION >= 2
+  addGSLDOUBLE2Func<gsl_sf_ellint_D>(SYM(D),SYM(phi),SYM(k));
+#else  
   addGSLDOUBLE3Func<gsl_sf_ellint_D>(SYM(D),SYM(phi),SYM(k),SYM(n));
+#endif  
   addGSLDOUBLE2Func<gsl_sf_ellint_RC>(SYM(RC),SYM(x),SYM(y));
   addGSLDOUBLE3Func<gsl_sf_ellint_RD>(SYM(RD),SYM(x),SYM(y),SYM(z));
   addGSLDOUBLE3Func<gsl_sf_ellint_RF>(SYM(RF),SYM(x),SYM(y),SYM(z));
