Without <sys/time.h>, gettimeofday is undefined.

diff --git a/src/AudioIO.cpp b/src/AudioIO.cpp
index 0187e3962..e15d55f4c 100644
--- a/src/AudioIO.cpp
+++ b/src/AudioIO.cpp
@@ -479,6 +479,8 @@ time warp info and AudioIOListener and whether the playback is looped.
 #include "../lib-src/portmidi/porttime/porttime.h"
 #include "../lib-src/header-substitutes/allegro.h"
 
+#include <sys/time.h>
+
    #define MIDI_SLEEP 10 /* milliseconds */
    // how long do we think the thread that fills MIDI buffers,
    // if it is separate from the portaudio thread,
