--- a/aws-cpp-sdk-core-tests/http/HttpClientTest.cpp
+++ b/aws-cpp-sdk-core-tests/http/HttpClientTest.cpp
@@ -92,12 +92,12 @@ TEST(HttpClientTest, TestRandomURLWithNoProxy)
     makeRandomHttpRequest(httpClient, false);
 }
 
-TEST(HttpClientTest, TestRandomURLWithProxy)
-{
-    ClientConfiguration configuration = makeClientConfigurationWithProxy();
-    auto httpClient = CreateHttpClient(configuration);
-    makeRandomHttpRequest(httpClient, true); // we expect it to try to use proxy that is invalid
-}
+//TEST(HttpClientTest, TestRandomURLWithProxy)
+//{
+//    ClientConfiguration configuration = makeClientConfigurationWithProxy();
+//    auto httpClient = CreateHttpClient(configuration);
+//    makeRandomHttpRequest(httpClient, true); // we expect it to try to use proxy that is invalid
+//}
 
 TEST(HttpClientTest, TestRandomURLWithProxyAndDeclaredAsNonProxyHost)
 {
@@ -119,14 +119,14 @@ TEST(HttpClientTest, TestRandomURLWithProxyAndDeclaredParentDomainAsNonProxyHost
     makeRandomHttpRequest(httpClient, false);
 }
 
-TEST(HttpClientTest, TestRandomURLWithProxyAndOtherDeclaredAsNonProxyHost)
-{
-    ClientConfiguration configuration = makeClientConfigurationWithProxy();
-    configuration.nonProxyHosts = Aws::Utils::Array<Aws::String>(1);
-    configuration.nonProxyHosts[0] = "http://test.non.filtered.aws";
-    auto httpClient = CreateHttpClient(configuration);
-    makeRandomHttpRequest(httpClient, true);
-}
+//TEST(HttpClientTest, TestRandomURLWithProxyAndOtherDeclaredAsNonProxyHost)
+//{
+//    ClientConfiguration configuration = makeClientConfigurationWithProxy();
+//    configuration.nonProxyHosts = Aws::Utils::Array<Aws::String>(1);
+//    configuration.nonProxyHosts[0] = "http://test.non.filtered.aws";
+//    auto httpClient = CreateHttpClient(configuration);
+//    makeRandomHttpRequest(httpClient, true);
+//}
 
 // TODO: Pending Fix on Windows.
 #if ENABLE_CURL_CLIENT
