diff --git a/Makefile b/Makefile
index edd2c4a..73b25ea 100644
--- a/Makefile
+++ b/Makefile
@@ -30,8 +30,8 @@ all: $(PROG) $(TEST_PROG)
 
 # Adjust $(HTSDIR) to point to your top-level htslib directory
 HTSDIR = htslib-1.4.1
-include $(HTSDIR)/htslib.mk
-include $(HTSDIR)/htslib_static.mk
+#include $(HTSDIR)/htslib.mk
+#include $(HTSDIR)/htslib_static.mk
 HTSLIB = $(HTSDIR)/libhts.a
 BGZIP  = $(HTSDIR)/bgzip
 TABIX  = $(HTSDIR)/tabix
@@ -73,7 +73,7 @@ GSL_LIBS       =
 ifdef USE_GPL
     EXTRA_CPPFLAGS += -DUSE_GPL
     OBJS += polysomy.o peakfit.o
-    GSL_LIBS = -lgsl -lcblas
+    GSL_LIBS = -lgsl -lgslcblas
 endif
 
 prefix      = /usr/local
@@ -217,7 +217,7 @@ test/test-regidx.o: test/test-regidx.c regidx.h
 test/test-regidx: test/test-regidx.o regidx.o $(HTSLIB)
 	$(CC) $(ALL_LDFLAGS) -o $@ $^ $(HTSLIB) -lpthread $(HTSLIB_LIBS) $(ALL_LIBS)
 
-bcftools: $(HTSLIB) $(OBJS)
+bcftools: $(OBJS)
 	$(CC) $(ALL_LDFLAGS) -o $@ $(OBJS) $(HTSLIB) -lpthread $(HTSLIB_LIBS) $(GSL_LIBS) $(ALL_LIBS)
 
 doc/bcftools.1: doc/bcftools.txt
