--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,10 +60,6 @@ ExternalProject_Add(BearSource
             -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
             -DENABLE_UNIT_TESTS:BOOL=${ENABLE_UNIT_TESTS}
             -DENABLE_FUNC_TESTS:BOOL=${ENABLE_FUNC_TESTS}
-        TEST_BEFORE_INSTALL
-            1
-        TEST_COMMAND
-            ctest # or `ctest -T memcheck`
         )
 
 # Run the functional tests
@@ -77,12 +73,8 @@ if (ENABLE_FUNC_TESTS)
                 -DCMAKE_INSTALL_LIBDIR:PATH=${CMAKE_INSTALL_LIBDIR}
                 -DCMAKE_INSTALL_BINDIR:PATH=${CMAKE_INSTALL_BINDIR}
                 -DSTAGED_INSTALL_PREFIX:PATH=${STAGED_INSTALL_PREFIX}
-            TEST_BEFORE_INSTALL
-                1
             INSTALL_COMMAND
                 ""
-            TEST_COMMAND
-                ctest --verbose
             )
 endif ()
 
