From 2a9b15aefb842a734637f3d230936ea1b7c60096 Mon Sep 17 00:00:00 2001
From: Philip Kime <Philip@kime.org.uk>
Date: Thu, 8 Nov 2018 22:02:09 +0100
Subject: [PATCH] Fix to address #239

---
 lib/Biber.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Biber.pm b/lib/Biber.pm
index 8b1f80a5..d97fca29 100644
--- a/lib/Biber.pm
+++ b/lib/Biber.pm
@@ -311,6 +311,8 @@ sub parse_ctrlfile {
     unless (eval {$checkbuf = File::Slurper::read_text($ctrl_file_path, 'latin1')}) {
       biber_error("$ctrl_file_path is not UTF-8 or even latin1, how horrible.");
     }
+    # Write ctrl file as UTF-8
+    File::Slurper::write_text($ctrl_file_path, NFC($checkbuf));# Unicode NFC boundary
   }
 
   $checkbuf = NFD($checkbuf);# Unicode NFD boundary
@@ -319,8 +321,6 @@ sub parse_ctrlfile {
     unlink($output) unless $output eq '-';# ignore deletion of STDOUT marker
     biber_error("$ctrl_file_path is malformed, last biblatex run probably failed. Deleted $output");
   }
-  # Write ctrl file as UTF-8
-  File::Slurper::write_text($ctrl_file_path, NFC($checkbuf));# Unicode NFC boundary
 
   # Validate if asked to
   if (Biber::Config->getoption('validate_control')) {
