Patch shebangs in source that gets unpacked by `configure'.

--- bigloo4.1a/gc/install-gc-7.4.0	2014-02-04 14:55:03.000000000 +0100
+++ bigloo4.1a/gc/install-gc-7.4.0	2014-02-04 14:55:36.000000000 +0100
@@ -29,10 +29,12 @@ fi
 
 # untar the two versions of the GC
 $tar xfz $src -C ../gc || (echo "$tar xfz $src failed"; exit 1)
-/bin/rm -rf "../gc/$gc"_fth
+rm -rf "../gc/$gc"_fth
+find ../gc/$gc -perm /111 -type f | xargs sed -i -e"s|/bin/sh|`type -P sh`|g"
 mv ../gc/$gc "../gc/$gc"_fth || (echo "mv $gc failed"; exit 1)
 
 $tar xfz $src  -C ../gc || (echo "$tar xfz $src failed"; exit 1)
+find ../gc/$gc -perm /111 -type f | xargs sed -i -e"s|/bin/sh|`type -P sh`|g"
 
 # general Bigloo patch
 (cd "../gc/$gc"_fth && $patch -p1 < ../$gc.patch > /dev/null)
