From 9edf67191cc3655480b6fd418247709ade930b1a Mon Sep 17 00:00:00 2001
From: Brendan Tildesley <mail@brendan.scot>
Date: Thu, 25 Feb 2021 00:33:10 +1100
Subject: [PATCH] Remove test_unrar

---
 src/calibre/test_build.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/calibre/test_build.py b/src/calibre/test_build.py
index b37fb1bcfb..0ab7aa0646 100644
--- a/src/calibre/test_build.py
+++ b/src/calibre/test_build.py
@@ -369,10 +369,6 @@ def test_file_dialog_helper(self):
         from calibre.gui2.win_file_dialogs import test
         test()

-    def test_unrar(self):
-        from calibre.utils.unrar import test_basic
-        test_basic()
-
     def test_7z(self):
         from calibre.utils.seven_zip import test_basic
         test_basic()
--
2.30.1
