--- a/clx.asd	2016-02-16 00:06:48.161596976 -0500
+++ b/clx.asd	2016-02-16 00:06:54.793774658 -0500
@@ -79,24 +79,6 @@
                (:file "xtest")
                (:file "screensaver")
                (:file "xinerama")))
-     (:module demo
-	      :default-component-class example-source-file
-	      :components
-	      ((:file "bezier")
-	       ;; KLUDGE: this requires "bezier" for proper operation,
-	       ;; but we don't declare that dependency here, because
-	       ;; asdf doesn't load example files anyway.
-	       (:file "beziertest")
-	       (:file "clclock")
-               (:file "clipboard")
-	       (:file "clx-demos")
-	       (:file "gl-test")
-	       ;; FIXME: compiling this generates 30-odd spurious code
-	       ;; deletion notes.  Find out why, and either fix or
-	       ;; workaround the problem.
-	       (:file "mandel")
-	       (:file "menu")
-	       (:file "zoid")))
      (:module test
 	      :default-component-class example-source-file
 	      :components
