From 96ceffdfdcedd3ae33c46b11357aa2518e0a6152 Mon Sep 17 00:00:00 2001
From: Petter <petter@mykolab.ch>
Date: Thu, 27 Apr 2017 18:55:10 +0200
Subject: [PATCH] Remove non-free fonts from settings menu

--- a/app/qml/ApplicationSettings.qml
+++ b/app/qml/ApplicationSettings.qml
@@ -409,7 +409,7 @@
             obj_string: '{"ambientLight":0.16,"backgroundColor":"#000000","bloom":0.4,"brightness":0.5,"flickering":0.1,"contrast":0.85,"fontName":"TERMINUS_SCALED","fontColor":"#0ccc68","frameName":"SIMPLE_WHITE_FRAME","glowingLine":0.2,"horizontalSync":0.16,"jitter":0.18,"burnIn":0.45,"staticNoise":0.1,"rasterization":0,"screenCurvature":0.1,"windowOpacity":1,"chromaColor":0,"saturationColor":0,"rbgShift":0,"fontWidth":1.0,"useCustomCommand":false,"customCommand":""}'
             builtin: true
         }
-        ListElement{
+        /*ListElement{
             text: "Default Scanlines"
             obj_string: '{"ambientLight":0.16,"backgroundColor":"#000000","bloom":0.4,"brightness":0.5,"flickering":0.1,"contrast":0.85,"fontName":"COMMODORE_PET","fontColor":"#00ff5b","frameName":"SIMPLE_WHITE_FRAME","glowingLine":0.2,"horizontalSync":0.14,"jitter":0.11,"burnIn":0.4,"staticNoise":0.05,"rasterization":1,"screenCurvature":0.1,"windowOpacity":1,"chromaColor":0,"saturationColor":0,"rbgShift":0,"fontWidth":1.0,"useCustomCommand":false,"customCommand":""}'
             builtin: true
@@ -433,7 +433,7 @@
             text: "IBM Dos"
             obj_string: '{"ambientLight":0.16,"backgroundColor":"#000000","bloom":0.4,"brightness":0.5,"flickering":0.07,"contrast":0.85,"fontName":"IBM_DOS","fontColor":"#ffffff","frameName":"SIMPLE_WHITE_FRAME","glowingLine":0.13,"horizontalSync":0,"jitter":0.16,"burnIn":0.3,"staticNoise":0.03,"rasterization":0,"screenCurvature":0.1,"windowOpacity":1,"chromaColor":1,"saturationColor":0,"rbgShift":0.35,"fontWidth":1.0,"useCustomCommand":false,"customCommand":""}'
             builtin: true
-        }
+        }*/
         ListElement{
             text: "IBM 3278"
             obj_string: '{"ambientLight":0.1,"backgroundColor":"#000000","bloom":0.15,"brightness":0.5,"flickering":0,"contrast":0.85,"fontName":"IBM_3278","fontColor":"#0ccc68","frameName":"SIMPLE_WHITE_FRAME","glowingLine":0,"horizontalSync":0,"jitter":0,"burnIn":0.6,"staticNoise":0,"rasterization":0,"screenCurvature":0.1,"windowOpacity":1,"chromaColor":0,"saturationColor":0,"rbgShift":0,"fontWidth":1.0,"useCustomCommand":false,"customCommand":""}'
diff --git a/app/qml/FontPixels.qml b/app/qml/FontPixels.qml
index 6ff85da..6f0d0ba 100644
--- a/app/qml/FontPixels.qml
+++ b/app/qml/FontPixels.qml
@@ -32,7 +32,7 @@ QtObject{
     property bool lowResolutionFont: true
 
     property ListModel fontlist: ListModel{
-        ListElement{
+        /*ListElement{
             name: "COMMODORE_PET"
             text: "Commodore PET (1977)"
             source: "fonts/1977-commodore-pet/COMMODORE_PET.ttf"
@@ -40,7 +40,7 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.8
-        }
+        }*/
         ListElement{
             name: "PROGGY_TINY"
             text: "Proggy Tiny (Modern)"
@@ -68,7 +68,7 @@ QtObject{
             baseScaling: 3.0
             fontWidth: 1.0
         }
-        ListElement{
+        /*ListElement{
             name: "APPLE_II"
             text: "Apple ][ (1977)"
             source: "fonts/1977-apple2/PrintChar21.ttf"
@@ -76,8 +76,8 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.9
-        }
-        ListElement{
+        }*/
+        /*ListElement{
             name: "ATARI_400"
             text: "Atari 400-800 (1979)"
             source: "fonts/1979-atari-400-800/ATARI400800_original.TTF"
@@ -85,8 +85,8 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.8
-        }
-        ListElement{
+        }*/
+        /*ListElement{
             name: "COMMODORE_64"
             text: "Commodore 64 (1982)"
             source: "fonts/1982-commodore64/C64_Pro_Mono_v1.0-STYLE.ttf"
@@ -94,6 +94,6 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.8
-        }
+        }*/
     }
 }
diff --git a/app/qml/FontScanlines.qml b/app/qml/FontScanlines.qml
index eebf00b..410d7b6 100644
--- a/app/qml/FontScanlines.qml
+++ b/app/qml/FontScanlines.qml
@@ -32,7 +32,7 @@ QtObject{
     property bool lowResolutionFont: true
 
     property ListModel fontlist: ListModel{
-        ListElement{
+        /*ListElement{
             name: "COMMODORE_PET"
             text: "Commodore PET (1977)"
             source: "fonts/1977-commodore-pet/COMMODORE_PET.ttf"
@@ -40,7 +40,7 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.7
-        }
+        }*/
         ListElement{
             name: "PROGGY_TINY"
             text: "Proggy Tiny (Modern)"
@@ -68,7 +68,7 @@ QtObject{
             baseScaling: 3.0
             fontWidth: 1.0
         }
-        ListElement{
+        /*ListElement{
             name: "APPLE_II"
             text: "Apple ][ (1977)"
             source: "fonts/1977-apple2/PrintChar21.ttf"
@@ -76,8 +76,8 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.8
-        }
-        ListElement{
+        }*/
+        /*ListElement{
             name: "ATARI_400"
             text: "Atari 400-800 (1979)"
             source: "fonts/1979-atari-400-800/ATARI400800_original.TTF"
@@ -85,8 +85,8 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.7
-        }
-        ListElement{
+        }*/
+        /*ListElement{
             name: "COMMODORE_64"
             text: "Commodore 64 (1982)"
             source: "fonts/1982-commodore64/C64_Pro_Mono_v1.0-STYLE.ttf"
@@ -94,6 +94,6 @@ QtObject{
             pixelSize: 8
             baseScaling: 4.0
             fontWidth: 0.7
-        }
+        }*/
     }
 }
diff --git a/app/qml/Fonts.qml b/app/qml/Fonts.qml
index ad20844..882808a 100644
--- a/app/qml/Fonts.qml
+++ b/app/qml/Fonts.qml
@@ -80,7 +80,7 @@
             fontWidth: 1.0
             lowResolutionFont: true
         }
-        ListElement{
+        /*ListElement{
             name: "COMMODORE_PET_SCALED"
             text: "Commodore PET (1977)"
             source: "fonts/1977-commodore-pet/COMMODORE_PET.ttf"
@@ -89,7 +89,7 @@
             baseScaling: 3.5
             fontWidth: 0.7
             lowResolutionFont: true
-        }
+        }*/
         ListElement{
             name: "PROGGY_TINY_SCALED"
             text: "Proggy Tiny (Modern)"
@@ -100,7 +100,7 @@
             fontWidth: 0.9
             lowResolutionFont: true
         }
-        ListElement{
+        /*ListElement{
             name: "APPLE_II_SCALED"
             text: "Apple ][ (1977)"
             source: "fonts/1977-apple2/PrintChar21.ttf"
@@ -149,7 +149,7 @@
             baseScaling: 2.0
             fontWidth: 1.0
             lowResolutionFont: true
-        }
+        }*/
         ListElement{
             name: "HERMIT"
             text: "HD: Hermit (Modern)"
@@ -177,7 +177,7 @@
             fontWidth: 1.0
             lowResolutionFont: false
         }
-        ListElement{
+        /*ListElement{
             name: "MONACO"
             text: "HD: Monaco (Modern)"
             source: "fonts/modern-monaco/monaco.ttf"
@@ -185,7 +185,7 @@
             pixelSize: 30
             fontWidth: 1.0
             lowResolutionFont: false
-        }
+        }*/
         ListElement{
             name: "INCONSOLATA"
             text: "HD: Inconsolata (Modern)"
