From c91d7ae5dbb00c8392a9f93283dc56c3e296cccd Mon Sep 17 00:00:00 2001
From: Petter <petter@mykolab.ch>
Date: Thu, 27 Apr 2017 20:19:21 +0200
Subject: [PATCH] Fix size of the array passed to memset()

---
 qmltermwidget/lib/History.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qmltermwidget/lib/History.cpp b/qmltermwidget/lib/History.cpp
index 0f9c13f..ab6f7be 100644
--- a/qmltermwidget/lib/History.cpp
+++ b/qmltermwidget/lib/History.cpp
@@ -515,7 +515,7 @@ void HistoryScrollBlockArray::addCells(const Character a[], int count)
   // put cells in block's data
   assert((count * sizeof(Character)) < ENTRIES);
 
-  memset(b->data, 0, ENTRIES);
+  memset(b->data, 0, sizeof(b->data));
 
   memcpy(b->data, a, count * sizeof(Character));
   b->size = count * sizeof(Character);
-- 
2.12.2

