From 08628fda19128b75248548357e416bc373f14f91 Mon Sep 17 00:00:00 2001
From: Yen Chi Hsuan <yan12125@gmail.com>
Date: Sat, 18 Mar 2017 02:50:34 +0800
Subject: [PATCH] Fix memory leak in hotspot (URLs & emails) detection

---
 qmltermwidget/lib/Filter.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/qmltermwidget/lib/Filter.cpp b/qmltermwidget/lib/Filter.cpp
index 5ca7bee..2e8d2fb 100644
--- a/qmltermwidget/lib/Filter.cpp
+++ b/qmltermwidget/lib/Filter.cpp
@@ -26,6 +26,7 @@
 // Qt
 #include <QAction>
 #include <QApplication>
+#include <QtAlgorithms>
 #include <QClipboard>
 #include <QString>
 #include <QTextStream>
@@ -194,6 +195,7 @@ Filter::~Filter()
 }
 void Filter::reset()
 {
+    qDeleteAll(_hotspotList);
     _hotspots.clear();
     _hotspotList.clear();
 }
-- 
2.12.2

