diff --git a/tests/data/test172 b/tests/data/test172
index b3efae9..3d53418 100644
--- a/tests/data/test172
+++ b/tests/data/test172
@@ -36,7 +36,7 @@ http://%HOSTIP:%HTTPPORT/we/want/172 -b log/jar172.txt -b "tool=curl; name=fool"
 
 .%HOSTIP	TRUE	/silly/	FALSE	0	ismatch	this
 .%HOSTIP	TRUE	/	FALSE	0	partmatch	present
-%HOSTIP	FALSE	/we/want/	FALSE	1391252187	nodomain	value
+%HOSTIP	FALSE	/we/want/	FALSE	2139150993	nodomain	value
 </file>
 </client>
