From 5676766be5e845ccb6cdf46cfa8722497f151752 Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Fri, 16 Jun 2017 15:11:37 -0400
Subject: Use 'ref' keyword for iter, requires vala 0.36


diff --git a/deja-dup/widgets/ConfigList.vala b/deja-dup/widgets/ConfigList.vala
index 15de2d6..02cd81a 100644
--- a/deja-dup/widgets/ConfigList.vala
+++ b/deja-dup/widgets/ConfigList.vala
@@ -333,7 +333,7 @@ public class ConfigList : ConfigWidget
 
     model.row_deleted.disconnect(write_to_config);
     foreach (Gtk.TreeIter iter in iters) {
-      (model as Gtk.ListStore).remove(iter);
+      (model as Gtk.ListStore).remove(ref iter);
     }
     model.row_deleted.connect(write_to_config);
 
diff --git a/deja-dup/widgets/ConfigLocation.vala b/deja-dup/widgets/ConfigLocation.vala
index 869e2a8..d21c556 100644
--- a/deja-dup/widgets/ConfigLocation.vala
+++ b/deja-dup/widgets/ConfigLocation.vala
@@ -397,12 +397,12 @@ public class ConfigLocation : ConfigWidget
     if (uuid == saved_uuid)
       return;
 
-    store.remove(iter);
+    store.remove(ref iter);
 
     if (--num_volumes == 0) {
       Gtk.TreeIter sep_iter;
       if (store.get_iter_from_string(out sep_iter, index_vol_sep.to_string())) {
-        store.remove(sep_iter);
+        store.remove(ref sep_iter);
         index_vol_sep = -2;
       }
     }
-- 
cgit v0.10.2

