From e5be173e201710538464c279335bef735a327a68 Mon Sep 17 00:00:00 2001
From: Brian Leung <bkleung89@gmail.com>
Date: Tue, 18 Jun 2019 06:37:46 +0200
Subject: [PATCH] Add missing require.

---
 unpackaged.el | 1 +
 1 file changed, 1 insertion(+)

diff --git a/unpackaged.el b/unpackaged.el
index 2d46b3c..e62184a 100644
--- a/unpackaged.el
+++ b/unpackaged.el
@@ -38,6 +38,7 @@
 (require 'dash)
 (require 's)
 (require 'use-package)
+(require 'hydra)
 
 ;;; Faces, fonts
 
-- 
2.22.0

