See http://sourceforge.net/p/flac/bugs/364/

diff -Naur flac-1.2.1-orig/examples/cpp/encode/file/main.cpp flac-1.2.1-ae/examples/cpp/encode/file/main.cpp
--- flac-1.2.1-orig/examples/cpp/encode/file/main.cpp	2012-12-27 20:15:11.000000000 +0100
+++ flac-1.2.1-ae/examples/cpp/encode/file/main.cpp	2012-12-27 20:25:01.000000000 +0100
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include "FLAC++/metadata.h"
 #include "FLAC++/encoder.h"
 
