Fixes a possible buffer overflow.  (There are others)
Author: Danny Milosavljevic <dannym@scratchpost.org>
Date: 2020-11-16

--- a/gcc/gimplify.c.orig	2020-10-29 08:45:11.532479774 +0100
+++ b/gcc/gimplify.c	2020-10-29 08:45:49.584637018 +0100
@@ -5229,7 +5229,7 @@
 	     flexibility, split it into separate input and output
  	     operands.  */
 	  tree input;
-	  char buf[10];
+	  char buf[12];
 
 	  /* Turn the in/out constraint into an output constraint.  */
 	  char *p = xstrdup (constraint);
