From 444250fce55f916af993bc855930c6809686e4bd Mon Sep 17 00:00:00 2001
From: Josselin Poiret <dev@jpoiret.xyz>
Date: Tue, 23 Nov 2021 18:39:39 +0000
Subject: [PATCH] Make GDM pass GDK_PIXBUF_MODULE_FILE to sessions

---
 daemon/gdm-launch-environment.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/daemon/gdm-launch-environment.c b/daemon/gdm-launch-environment.c
index 14ecfac2..3e618321 100644
--- a/daemon/gdm-launch-environment.c
+++ b/daemon/gdm-launch-environment.c
@@ -158,6 +158,7 @@ build_launch_environment (GdmLaunchEnvironment *launch_environment,
                 "WINDOWPATH",
                 "XCURSOR_PATH",
                 "XDG_CONFIG_DIRS",
+                "GDK_PIXBUF_MODULE_FILE",
                 NULL
         };
         char *system_data_dirs;
-- 
2.33.1

