diff --git a/test/mke2fs.dump b/test/mke2fs.dump
index 8e63662..0c25798 100644
--- a/test/mke2fs.0.dump
+++ b/test/mke2fs.0.dump
@@ -11,7 +11,7 @@
 Inode count:              8192
 Block count:              32768
 Reserved block count:     1638
-Free blocks:              26568
+Free blocks:              26566
 Free inodes:              8141
 First block:              1
 Block size:               1024
@@ -29,7 +29,7 @@
 Maximum mount count:      -1
 Last checked:             Sat Jan  1 00:00:00 2000
 Check interval:           0 (<none>)
-Lifetime writes:          141 kB
+Lifetime writes:          143 kB
 Reserved blocks uid:      0 (user root)
 Reserved blocks gid:      0 (group root)
 First inode:              11
@@ -51,13 +51,13 @@
 Journal start:            0
 
 
-Group 0: (Blocks 1-8192) csum 0x74a0 [ITABLE_ZEROED]
+Group 0: (Blocks 1-8192) csum 0x3f30 [ITABLE_ZEROED]
   Primary superblock at 1, Group descriptors at 2-2
-  Block bitmap at 3 (+2), csum 0x16cec4db
+  Block bitmap at 3 (+2), csum 0x8d67cdef
   Inode bitmap at 7 (+6), csum 0xb1052088
   Inode table at 11-522 (+10)
-  6093 free blocks, 1997 free inodes, 18 directories, 1997 unused inodes
-  Free blocks: 2100-8192
+  6091 free blocks, 1997 free inodes, 18 directories, 1997 unused inodes
+  Free blocks: 2102-8192
   Free inodes: 52-2048
 Group 1: (Blocks 8193-16384) csum 0x8fde [INODE_UNINIT, BLOCK_UNINIT, ITABLE_ZEROED]
   Backup superblock at 8193, Group descriptors at 8194-8194
