Default to the pinentry program installed in ~/.guix-profile.

diff --git a/common/homedir.c b/common/homedir.c
index e9e75d0..74e0aaf 100644
--- a/common/homedir.c
+++ b/common/homedir.c
@@ -968,7 +968,7 @@ get_default_pinentry_name (int reset)
   } names[] = {
     /* The first entry is what we return in case we found no
        other pinentry.  */
-    { gnupg_bindir, DIRSEP_S "pinentry" EXEEXT_S },
+    { gnupg_homedir, "/.guix-profile/bin/pinentry" },
 #ifdef HAVE_W32_SYSTEM
     /* Try Gpg4win directory (with bin and without.) */
     { w32_rootdir, "\\..\\Gpg4win\\bin\\pinentry.exe" },
