Remove duplicate manpage entries from Makefile.

--- gnutls-3.4.0/doc/manpages/Makefile.am.orig	2015-04-06 04:48:30.000000000 -0400
+++ gnutls-3.4.0/doc/manpages/Makefile.am	2015-04-12 16:52:58.029694525 -0400
@@ -134,11 +134,8 @@
 APIMANS += gnutls_certificate_get_peers_subkey_id.3
 APIMANS += gnutls_certificate_get_trust_list.3
 APIMANS += gnutls_certificate_get_verify_flags.3
-APIMANS += gnutls_certificate_get_verify_flags.3
-APIMANS += gnutls_certificate_get_x509_crt.3
 APIMANS += gnutls_certificate_get_x509_crt.3
 APIMANS += gnutls_certificate_get_x509_key.3
-APIMANS += gnutls_certificate_get_x509_key.3
 APIMANS += gnutls_certificate_send_x509_rdn_sequence.3
 APIMANS += gnutls_certificate_server_set_request.3
 APIMANS += gnutls_certificate_set_dh_params.3
--- gnutls-3.4.0/doc/manpages/Makefile.in.orig	2015-04-08 02:08:30.000000000 -0400
+++ gnutls-3.4.0/doc/manpages/Makefile.in	2015-04-12 16:53:13.319694530 -0400
@@ -1275,11 +1275,8 @@
 	gnutls_certificate_get_peers_subkey_id.3 \
 	gnutls_certificate_get_trust_list.3 \
 	gnutls_certificate_get_verify_flags.3 \
-	gnutls_certificate_get_verify_flags.3 \
-	gnutls_certificate_get_x509_crt.3 \
 	gnutls_certificate_get_x509_crt.3 \
 	gnutls_certificate_get_x509_key.3 \
-	gnutls_certificate_get_x509_key.3 \
 	gnutls_certificate_send_x509_rdn_sequence.3 \
 	gnutls_certificate_server_set_request.3 \
 	gnutls_certificate_set_dh_params.3 \
