--- a/giscanner/sourcescanner.py	2013-12-15 23:03:54.002937000 +0100
+++ b/giscanner/sourcescanner.py	2013-12-15 23:04:50.322937000 +0100
@@ -277,6 +277,11 @@
         defines = ['__GI_SCANNER__']
         undefs = []
         cpp_args = os.environ.get('CC', 'cc').split()  # support CC="ccache gcc"
+        if (cpp_args == ['cc'] and
+            not any(map(lambda x: os.access(os.path.join(x, 'cc'), os.X_OK),
+                        os.environ.get('PATH').split(':')))):
+            cpp_args = ['GUIX_GCC_PATH']
+
         if 'cl' in cpp_args:
             # The Microsoft compiler/preprocessor (cl) does not accept
             # source input from stdin (the '-' flag), so we need
