From 0466df8660a7b6ddf2e082a1ec38bc6ea25a3c5c Mon Sep 17 00:00:00 2001
From: Patrick Alken <alken@colorado.edu>
Date: Mon, 7 Apr 2014 10:59:58 -0600
Subject: [PATCH] change error test for 15th degree polynomial (bug #39055)

---
 poly/test.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/poly/test.c b/poly/test.c
index d090802..f6a7e3f 100644
--- a/poly/test.c
+++ b/poly/test.c
@@ -579,8 +579,8 @@ main (void)
 
     for (i = 0; i<15; i++)
       {
-        gsl_test_abs (z[2*i], expected[2*i], 1e-7, "z%d.real, 15th-order polynomial", i);
-        gsl_test_abs (z[2*i+1], expected[2*i+1], 1e-7, "z%d.imag, 15th-order polynomial", i);
+        gsl_test_rel (z[2*i], expected[2*i], 1e-7, "z%d.real, 15th-order polynomial", i);
+        gsl_test_rel (z[2*i+1], expected[2*i+1], 1e-7, "z%d.imag, 15th-order polynomial", i);
       }
   }
 
-- 
2.4.3

