From: Tobias Geerinckx-Rice <me@tobias.gr>
Date: Sun, 04 Oct 2020 13:28:49 +0200
Subject: [PATCH] gnu: hplip: Fix non-network builds (bug #1898438)

Reported as <https://bugs.launchpad.net/hplip/+bug/1898438>.

diff -Naur a/scan/sane/hpaio.c b/scan/sane/hpaio.c
--- a/scan/sane/hpaio.c	1970-01-01 01:00:01.000000000 +0100
+++ b/scan/sane/hpaio.c	2020-10-04 13:26:34.665244052 +0200
@@ -36,7 +36,9 @@
 #include <string.h>
 #include <cups/cups.h>
 #include "hpmud.h"
+#ifdef HAVE_LIBNETSNMP
 #include "avahiDiscovery.h"
+#endif
 #include "hp_ipp.h"
 #include "soap.h"
 #include "soapht.h"
