Fixes doc++'s treatment of tabular within a parameter block.

From commit 883925f8a at http://github.com/LLNL/hypre

--- hypre-2.10.1/src/parcsr_ls/HYPRE_parcsr_ls.h	2015-12-04 22:12:19.000000000 -0600
+++ hypre-2.10.1/src/parcsr_ls/HYPRE_parcsr_ls.h	2016-03-16 09:02:58.547501336 -0500
@@ -1154,8 +1154,6 @@
  * Set the symmetry parameter for the
  * ParaSails preconditioner.
  *
- * @param solver [IN] Preconditioner object for which to set symmetry parameter.
- * @param sym [IN] Value of the symmetry parameter:
  * \begin{tabular}{|c|l|} \hline 
  * value & meaning \\ \hline 
  * 0 & nonsymmetric and/or indefinite problem, and nonsymmetric preconditioner\\
@@ -1163,6 +1161,9 @@
  * 2 & nonsymmetric, definite problem, and SPD (factored) preconditioner \\
  * \hline
  * \end{tabular}
+ * 
+ * @param solver [IN] Preconditioner object for which to set symmetry parameter.
+ * @param sym [IN] Value of the symmetry parameter:
  **/
 HYPRE_Int HYPRE_ParaSailsSetSym(HYPRE_Solver solver,
                                 HYPRE_Int    sym);
