Fixes needed when avoiding bundled libraries.

--- icecat-60.5.0/xpcom/build/moz.build.orig	2018-09-13 17:46:49.000000000 -0400
+++ icecat-60.5.0/xpcom/build/moz.build	2018-09-22 04:26:50.659564554 -0400
@@ -99,10 +99,5 @@
     '/docshell/base',
 ]
 
-if CONFIG['MOZ_VPX']:
-    LOCAL_INCLUDES += [
-        '/media/libvpx',
-    ]
-
 if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'cocoa':
     CXXFLAGS += CONFIG['TK_CFLAGS']
--- icecat-60.5.0/storage/moz.build.orig	2018-09-13 17:51:11.000000000 -0400
+++ icecat-60.5.0/storage/moz.build	2018-09-22 04:26:50.659564554 -0400
@@ -117,7 +117,6 @@
     DEFINES['MOZ_MEMORY_TEMP_STORE_PRAGMA'] = True
 
 LOCAL_INCLUDES += [
-    '/db/sqlite3/src',
     '/dom/base',
 ]
 
--- icecat-60.5.0/dom/indexedDB/moz.build.orig	2018-09-13 17:49:42.000000000 -0400
+++ icecat-60.5.0/dom/indexedDB/moz.build	2018-09-22 04:26:50.663564574 -0400
@@ -102,7 +102,6 @@
     CXXFLAGS += ['-Wno-error=shadow']
 
 LOCAL_INCLUDES += [
-    '/db/sqlite3/src',
     '/dom/base',
     '/dom/storage',
     '/ipc/glue',
--- icecat-60.5.0/media/webrtc/trunk/webrtc/base/rtc_task_queue_gn/moz.build.orig	2018-09-13 17:40:54.000000000 -0400
+++ icecat-60.5.0/media/webrtc/trunk/webrtc/base/rtc_task_queue_gn/moz.build	2018-09-23 21:33:12.319975105 -0400
@@ -130,11 +130,6 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/ipc/chromium/src/third_party/libevent/include/",
-        "/ipc/chromium/src/third_party/libevent/linux/"
-    ]
-
     UNIFIED_SOURCES += [
         "/media/webrtc/trunk/webrtc/base/task_queue_libevent.cc",
         "/media/webrtc/trunk/webrtc/base/task_queue_posix.cc"
