Fixes needed when avoiding bundled libraries.

--- icecat-52.0.2/xpcom/build/moz.build.orig
+++ icecat-52.0.2/xpcom/build/moz.build
@@ -93,10 +93,5 @@
     '/docshell/base',
 ]
 
-if CONFIG['MOZ_VPX']:
-    LOCAL_INCLUDES += [
-        '/media/libvpx',
-    ]
-
 if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'cocoa':
     CXXFLAGS += CONFIG['TK_CFLAGS']
--- icecat-52.0.2/storage/moz.build.orig
+++ icecat-52.0.2/storage/moz.build
@@ -114,7 +114,6 @@
     DEFINES['MOZ_MEMORY_TEMP_STORE_PRAGMA'] = True
 
 LOCAL_INCLUDES += [
-    '/db/sqlite3/src',
     '/dom/base',
 ]
 
--- icecat-52.0.2/dom/indexedDB/moz.build.orig
+++ icecat-52.0.2/dom/indexedDB/moz.build
@@ -101,7 +101,6 @@
     CXXFLAGS += ['-Wno-error=shadow']
 
 LOCAL_INCLUDES += [
-    '/db/sqlite3/src',
     '/dom/base',
     '/dom/storage',
     '/dom/workers',
