From 0ae15d7e89b984f72780ff7afa8e57dd6ae7765a Mon Sep 17 00:00:00 2001
From: "Boris A. Dekshteyn" <boris.dekshteyn@gmail.com>
Date: Mon, 14 Jun 2021 23:52:16 +1200
Subject: [PATCH] Fix build witch GCC@7.5.

---
 src/object/sp-use-reference.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/object/sp-use-reference.h b/src/object/sp-use-reference.h
index 5958632059..f07c2a3721 100644
--- a/src/object/sp-use-reference.h
+++ b/src/object/sp-use-reference.h
@@ -14,6 +14,7 @@
 
 #include "sp-item.h"
 #include "uri-references.h"
+#include "display/curve.h"
 
 #include <memory>
 
-- 
2.32.0

