diff --git a/std/socket.d b/std/socket.d
index 111cd17..6d23da9 100644
--- a/std/socket.d
+++ b/std/socket.d
@@ -448,7 +448,7 @@ class Protocol
 // Skip this test on Android because getprotobyname/number are
 // unimplemented in bionic.
 version(CRuntime_Bionic) {} else
-@safe unittest
+@safe version(hasNetwork) unittest
 {
     // import std.stdio : writefln;
     softUnittest({
@@ -770,7 +770,7 @@ class InternetHost
 }
 
 ///
-@safe unittest
+@safe version(hasNetwork) unittest
 {
     InternetHost ih = new InternetHost;
 
