Adapted from https://github.com/digego/extempore/pull/322/files

--- a/include/llvm/IR/ValueMap.h	2015-08-04 00:30:24.000000000 +0200
+++ b/include/llvm/IR/ValueMap.h	2018-07-14 21:09:09.769502736 +0200
@@ -99,7 +99,7 @@
   explicit ValueMap(const ExtraData &Data, unsigned NumInitBuckets = 64)
       : Map(NumInitBuckets), Data(Data) {}

-  bool hasMD() const { return MDMap; }
+  bool hasMD() const { return static_cast<bool>(MDMap); }
   MDMapT &MD() {
     if (!MDMap)
       MDMap.reset(new MDMapT);
