See https://www.lyx.org/trac/ticket/10800#comment:17
diff --git a/src/tex2lyx/test/test-structure.lyx.lyx b/src/tex2lyx/test/test-structure.lyx.lyx
index feff755cd2..2c7f2ace5c 100644
--- a/src/tex2lyx/test/test-structure.lyx.lyx
+++ b/src/tex2lyx/test/test-structure.lyx.lyx
@@ -212,7 +212,7 @@ This causes the
 \begin_inset Flex Flex:Strong
 status collapsed
 
-\begin_layout Standard
+\begin_layout Plain Layout
 logikalmkup
 \end_layout
