From a83702834938b23cc2e843589aa223e2024a7e6f Mon Sep 17 00:00:00 2001
From: Orion Poplawski <orion@cora.nwra.com>
Date: Tue, 29 Nov 2016 11:48:01 -0700
Subject: [PATCH] Add missing #include "err_macros.h" to tst_h_par.c

---
 h5_test/tst_h_par.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/h5_test/tst_h_par.c b/h5_test/tst_h_par.c
index c3da7f4..a419d55 100644
--- a/h5_test/tst_h_par.c
+++ b/h5_test/tst_h_par.c
@@ -11,6 +11,7 @@
    $Id: tst_h_par.c,v 1.15 2010/05/25 13:53:04 ed Exp $
 */
 #include <nc_tests.h>
+#include "err_macros.h"
 #include <hdf5.h>
 
 /* Defining USE_MPE causes the MPE trace library to be used (and you
