From d3026a6d331298003ccc6cd9d2e20dcb7fa9ae1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20=C4=8Cech?= <sleep_walker@gnu.org>
Date: Wed, 3 Jul 2019 13:31:54 +0200
Subject: [PATCH] respect NM_VPN_PLUGIN_DIR

--- a/src/core/vpn/nm-vpn-manager.c	2021-10-29 13:20:01.062917840 -0400
+++ b/src/core/vpn/nm-vpn-manager.c	2021-10-29 13:26:46.094397018 -0400
@@ -211,6 +211,7 @@
     GSList *             infos, *info;
     const char *         conf_dir_etc = _nm_vpn_plugin_info_get_default_dir_etc();
     const char *         conf_dir_lib = _nm_vpn_plugin_info_get_default_dir_lib();
+    const char *         conf_dir_user = _nm_vpn_plugin_info_get_default_dir_user ();
 
     /* Watch the VPN directory for changes */
     file              = g_file_new_for_path(conf_dir_lib);
@@ -229,6 +230,14 @@
             g_signal_connect(priv->monitor_etc, "changed", G_CALLBACK(vpn_dir_changed), self);
     }
 
+    file              = g_file_new_for_path (conf_dir_user);
+    priv->monitor_etc = g_file_monitor_directory (file, G_FILE_MONITOR_NONE, NULL, NULL);
+    g_object_unref (file);
+    if (priv->monitor_etc) {
+        priv->monitor_id_etc =
+            g_signal_connect (priv->monitor_etc, "changed", G_CALLBACK (vpn_dir_changed), self);
+    }
+
     /* first read conf_dir_lib. The name files are not really user configuration, but
      * plugin configuration. Hence we expect ~newer~ plugins to install their files
      * in /usr/lib/NetworkManager. We want to prefer those files.
@@ -243,6 +252,11 @@
         try_add_plugin(self, info->data);
     g_slist_free_full(infos, g_object_unref);
 
+    infos = _nm_vpn_plugin_info_list_load_dir(conf_dir_user, TRUE, 0, NULL, NULL);
+    for (info = infos; info; info = info->next)
+            try_add_plugin (self, info->data);
+    g_slist_free_full (infos, g_object_unref);
+
     priv->active_services = g_hash_table_new_full(nm_str_hash, g_str_equal, g_free, NULL);
 }
 
