From 354ef78aac0b887fae3c10b28eb2b0d83f66bdfe Mon Sep 17 00:00:00 2001
From: Julien Lepiller <julien@lepiller.eu>
Date: Mon, 2 Jan 2017 17:05:24 +0100
Subject: [PATCH] Honor SOURCE_DATE_EPOCH

---
 Makefile.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index a32b4b8..ef4c174 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -113,11 +113,16 @@ graph.cmx: $(CMI) $(CMX)
	$(OCAMLOPT) $(INCLUDES) -pack -o $@ $^
 
 VERSION=1.8.7
+ifdef SOURCE_DATE_EPOCH
+BUILD_DATE=$(shell date -u -d "@$(SOURCE_DATE_EPOCH)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" 2>/dev/null || date)
+else
+BUILD_DATE=$(shell date)
+endif
 
 src/version.ml: Makefile
	rm -f $@
	echo "let version = \""$(VERSION)"\"" > $@
-	echo "let date = \""`date`"\"" >> $@
+	echo "let date = \""$(BUILD_DATE)"\"" >> $@
 
 # gtk2 graph editor
 ###################
-- 
2.11.0

