From e2b39dd3a0742853e00f9c3e8c46c911da20bed7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Adrien=20B=C3=A9raud?= <adrien.beraud@savoirfairelinux.com>
Date: Tue, 30 Jun 2020 10:42:49 -0400
Subject: [PATCH 1/4] http/request: make terminate public

---
 include/opendht/http.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/opendht/http.h b/include/opendht/http.h
index cc8d5f9..46b722c 100644
--- a/include/opendht/http.h
+++ b/include/opendht/http.h
@@ -294,6 +294,7 @@ public:
      * User action to cancel the Request and call the completion callbacks.
      */
     void cancel();
+    void terminate(const asio::error_code& ec);
 
 private:
     using OnCompleteCb = std::function<void()>;
@@ -320,8 +321,6 @@ private:
 
     void connect(std::vector<asio::ip::tcp::endpoint>&& endpoints, HandlerCb cb = {});
 
-    void terminate(const asio::error_code& ec);
-
     void post();
 
     void handle_request(const asio::error_code& ec);
-- 
2.27.0

