diff --git a/data/meson.build b/data/meson.build
index 46f6373..26a2c18 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -35,7 +35,6 @@ if get_option('profile') == 'development'
 endif
 desktop_conf.set('ICON', application_id)
 desktop = i18n.merge_file(
-  'desktop',
   input: configure_file(
     input: files('org.gnome.PasswordSafe.desktop.in.in'),
     output: 'org.gnome.PasswordSafe.desktop.in',
@@ -65,7 +64,6 @@ if get_option('profile') == 'development'
 endif
 appdata_conf.set('APPID', application_id)
 appdata = i18n.merge_file(
-  'appdata',
   input: configure_file(
     input: files('org.gnome.PasswordSafe.appdata.xml.in.in'),
     output: 'org.gnome.PasswordSafe.appdata.xml.in',
