Work around this GCC ICE: <https://bugs.gnu.org/31708>.

--- linux-4.16.13/tools/perf/util/header.c	2018-06-04 11:30:39.368146035 +0200
+++ linux-4.16.13/tools/perf/util/header.c	2018-06-04 11:34:04.667212378 +0200
@@ -135,7 +135,7 @@ int do_write(struct feat_fd *ff, const v
 int write_padded(struct feat_fd *ff, const void *bf,
 		 size_t count, size_t count_aligned)
 {
-	static const char zero_buf[NAME_ALIGN];
+	static const char zero_buf[NAME_ALIGN] = { 0 };
 	int err = do_write(ff, bf, count);
 
 	if (!err)
