Description: Add missing inlcudes of cstddef, fixing FTBFS13.1
Author: Regis Boudin <regis@debian.org>
Bug-Debian: http://bugs.debian.org/624953
Bug-Debian: http://bugs.debian.org/640726

--- pinball-0.3.1/base/Behavior.cpp
+++ pinball-0.3.1/base/Behavior.cpp
@@ -10,6 +10,8 @@
 #include "Behavior.h"
 #include "Group.h"
 
+#include <cstddef>
+
 Behavior::Behavior() {
 	//EM_COUT("Behavior::Behavior", 1);
 	p_Light = NULL;
--- pinball-0.3.1/base/SignalSender.cpp
+++ pinball-0.3.1/base/SignalSender.cpp
@@ -12,6 +12,8 @@
 #include "Group.h"
 #include "Behavior.h"
 
+#include <cstddef>
+
 SignalSender * SignalSender::p_SignalSender = NULL;
 
 SignalSender::SignalSender() {
--- pinball-0.3.1/base/BehaviorVisitor.cpp
+++ pinball-0.3.1/base/BehaviorVisitor.cpp
@@ -10,6 +10,8 @@
 #include "Behavior.h"
 #include "Group.h"
 
+#include <cstddef>
+
 BehaviorVisitor * BehaviorVisitor::p_BehaviorVisitor = NULL;
 
 BehaviorVisitor::BehaviorVisitor() {
