diff --git a/setup.py b/setup.py
index 4800173..6bdd77f 100755
--- a/setup.py
+++ b/setup.py
@@ -14,8 +14,7 @@ url = 'https://github.com/un33k/python-slugify'
 author = 'Val Neekman'
 author_email = 'info@neekware.com'
 license = 'MIT'
-install_requires = ['text-unidecode>=1.3']
-extras_require = {'unidecode': ['Unidecode>=1.1.1']}
+install_requires = ['Unidecode']
 
 classifiers = [
     'Development Status :: 5 - Production/Stable',
@@ -67,7 +66,6 @@ setup(
     author_email=author_email,
     packages=find_packages(exclude=EXCLUDE_FROM_PACKAGES),
     install_requires=install_requires,
-    extras_require=extras_require,
     classifiers=classifiers,
     entry_points={'console_scripts': ['slugify=slugify.slugify:main']},
 )
