--- Lib/test/test_posixpath.py  2014-03-01 05:46:56.984311000 +0100
+++ Lib/test/test_posixpath.py  2014-03-07 00:59:20.888311000 +0100
@@ -319,7 +319,11 @@
                 del env['HOME']
                 home = pwd.getpwuid(os.getuid()).pw_dir
                 # $HOME can end with a trailing /, so strip it (see #17809)
-                self.assertEqual(posixpath.expanduser("~"), home.rstrip("/"))
+                # The Guix builders have '/' as a home directory, so
+                # home.rstrip("/") will be an empty string and the test will
+                # fail. Let's just disable it since it does not really make
+                # sense with such a bizarre setup.
+                # self.assertEqual(posixpath.expanduser("~"), home.rstrip("/"))
