From 32bfaa7cc1c736fd62fcbb6414de9498dc20ed07 Mon Sep 17 00:00:00 2001
From: Adriano Peluso <catonano@gmail.com>
Date: Wed, 5 Apr 2017 15:13:06 +0200
Subject: [PATCH 2/2] buildable on python27 too

---
 genshi/template/directives.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/genshi/template/directives.py b/genshi/template/directives.py
index 6fd0f28..1f70ef6 100644
--- a/genshi/template/directives.py
+++ b/genshi/template/directives.py
@@ -266,7 +266,7 @@ class DefDirective(Directive):
         if isinstance(ast, _ast.Call):
             self.name = ast.func.id
             for arg in ast.args:
-                if isinstance(arg, _ast.Starred):
+                if hasattr(_ast, 'Starred') and isinstance(arg, _ast.Starred):
                     # Python 3.5+
                     self.star_args = arg.value.id
                 else:
-- 
2.12.0

