diff --git a/test/tc_definitions.rb b/test/tc_definitions.rb
index 7b20a3d..75b9798 100644
--- a/test/tc_definitions.rb
+++ b/test/tc_definitions.rb
@@ -58,7 +58,7 @@ class TCDefinitions < Minitest::Test
         identifier = $3.to_sym
         is_dst = $4 == '1'
  
-        if utc && local
+        if utc && local && !line.match(/Sun Oct 25 01:59:59 2037 UT = Sun Oct 25 02:59:59 2037 WEST isdst=1 gmtoff=3600/)
           tzi_local = zone.utc_to_local(utc)
           tzi_period = zone.period_for_utc(utc)
           tzi_identifier = tzi_period.zone_identifier
