Index: SDL_Pango-0.1.2/src/SDL_Pango.c
===================================================================
--- SDL_Pango-0.1.2.orig/src/SDL_Pango.c	2007-09-18 15:12:20.736253215 +0200
+++ SDL_Pango-0.1.2/src/SDL_Pango.c	2007-09-18 15:12:44.621614364 +0200
@@ -234,7 +234,7 @@
 //! non-zero if initialized
 static int IS_INITIALIZED = 0;
 
-#define DEFAULT_FONT_FAMILY "Sans"
+#define DEFAULT_FONT_FAMILY "sans-serif"
 #define DEFAULT_FONT_SIZE 12
 #define DEFAULT_DPI 96
 #define _MAKE_FONT_NAME(family, size) family " " #size
