--- a/src/SDL_Pango.h
+++ b/src/SDL_Pango.h
@@ -171,7 +171,7 @@
     SDLPango_Direction direction);
 
 
-#ifdef __FT2_BUILD_UNIX_H__
+#ifdef FT2BUILD_H_
 
 extern DECLSPEC void SDLCALL SDLPango_CopyFTBitmapToSurface(
     const FT_Bitmap *bitmap,
@@ -179,7 +179,7 @@
     const SDLPango_Matrix *matrix,
     SDL_Rect *rect);
 
-#endif	/* __FT2_BUILD_UNIX_H__ */
+#endif
 
 
 #ifdef __PANGO_H__
