From 49d83e24a8e66977056fc9920812265c16806500 Mon Sep 17 00:00:00 2001
From: carolili <carolili@iki.fi>
Date: Thu, 9 Feb 2017 19:24:49 +0000
Subject: [PATCH] Removing contribs

---
 configure.ac | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1cf1051..5d76b44 100644
--- a/configure.ac
+++ b/configure.ac
@@ -435,28 +435,6 @@ dnl All slurm Makefiles:
 
 AC_CONFIG_FILES([Makefile
 		 auxdir/Makefile
-		 contribs/Makefile
-		 contribs/cray/Makefile
-		 contribs/cray/csm/Makefile
-		 contribs/lua/Makefile
-		 contribs/mic/Makefile
-		 contribs/pam/Makefile
-		 contribs/pam_slurm_adopt/Makefile
-		 contribs/perlapi/Makefile
-		 contribs/perlapi/libslurm/Makefile
-		 contribs/perlapi/libslurm/perl/Makefile.PL
-		 contribs/perlapi/libslurmdb/Makefile
-		 contribs/perlapi/libslurmdb/perl/Makefile.PL
-		 contribs/seff/Makefile
-		 contribs/torque/Makefile
-		 contribs/openlava/Makefile
-		 contribs/phpext/Makefile
-		 contribs/phpext/slurm_php/config.m4
-		 contribs/sgather/Makefile
-		 contribs/sgi/Makefile
-		 contribs/sjobexit/Makefile
-		 contribs/slurmdb-direct/Makefile
-		 contribs/pmi2/Makefile
 		 doc/Makefile
 		 doc/man/Makefile
 		 doc/man/man1/Makefile
-- 
2.11.0

