From 5fb947886bf1e40475edf930fb2b284c24c8cb9a Mon Sep 17 00:00:00 2001
From: Sanel Zukan <karijes@users.sourceforge.net>
Date: Sun, 23 Dec 2018 00:04:34 +0100
Subject: [PATCH] Make sure INT_MAX32 is visible.

---
 src/thd_trip_point.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/thd_trip_point.h b/src/thd_trip_point.h
index 0eedc95..89eb083 100644
--- a/src/thd_trip_point.h
+++ b/src/thd_trip_point.h
@@ -29,6 +29,10 @@
 #include "thd_sys_fs.h"
 #include "thd_preference.h"
 #include "thd_cdev.h"
+
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+
 #include <time.h>
 #include <vector>
 #include <algorithm>    // std::sort
