--- totem-3.20.1/src/backend/bacon-video-widget.c.orig	2016-03-22 12:45:32.000000000 -0400
+++ totem-3.20.1/src/backend/bacon-video-widget.c	2016-05-11 02:49:55.299109551 -0400
@@ -2334,7 +2334,7 @@
     if (!gst_toc_entry_get_start_stop_times (entry, &start, &stop)) {
       GST_DEBUG ("Chapter #%d (couldn't get times)", i);
     } else {
-      GST_DEBUG ("Chapter #%d (start: %li stop: %li)", i, start, stop);
+      GST_DEBUG ("Chapter #%d (start: %li stop: %li)", i, (long) start, (long) stop);
     }
   }
 
