Accept glibc 2.19 as valid.

--- a/configure	2013-10-10 22:27:20.331223000 +0200
+++ b/configure	2013-10-10 22:27:55.055223000 +0200
@@ -6604,6 +6604,16 @@
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.19)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.19 family" >&5
+$as_echo "2.19 family" >&6; }
+
+$as_echo "#define GLIBC_2_18 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Darwin" >&5
 $as_echo "Darwin" >&6; }
