Description: Avoid __DATE__ and __TIME__ to build reproducibly
Author: Stephen Kitt <skitt@debian.org>

--- a/src/GameInit.cpp
+++ b/src/GameInit.cpp
@@ -248,7 +248,6 @@
   Logger::setActiv(XMSession::instance()->noLog() == false); /* apply log activ mode */
 
   LogInfo(std::string("X-Moto " + XMBuild::getVersionString(true)).c_str());
-  LogInfo("compiled at "__DATE__" "__TIME__);
   if(SwapEndian::bigendien) {
     LogInfo("Systeme is bigendien");
   } else {
--- a/src/states/StateOptions.cpp
+++ b/src/states/StateOptions.cpp
@@ -1233,8 +1233,6 @@
   int p=25;
   makeWindowOptions_infos_line(v_window, "Version", "X-Moto " + XMBuild::getVersionString(true), p);
   p+=20;
-  makeWindowOptions_infos_line(v_window, "Compilation date", __DATE__ " " __TIME__, p);
-  p+=20;
   makeWindowOptions_infos_line(v_window, "User data directory", XMFS::getUserDir(FDT_DATA), p);
   p+=20;
   makeWindowOptions_infos_line(v_window, "User config directory", XMFS::getUserDir(FDT_CONFIG), p);
