Description: Allow building without upstream-supplied glext.h
Author: Stephen Kitt <steve@sk2.org>
Bug: http://todo.xmoto.tuxfamily.org/index.php?do=details&task_id=803

--- xmoto-0.5.9+dfsg.orig/src/include/xm_OpenGL.h
+++ xmoto-0.5.9+dfsg/src/include/xm_OpenGL.h
@@ -1,7 +1,6 @@
 #ifdef ENABLE_OPENGL
 /* Pull in OpenGL headers */
 /* following scissored from SDL_opengl.h */
-#define __glext_h_  /* Don't let gl.h include glext.h */
 #ifdef HAVE_APPLE_OPENGL_FRAMEWORK
 #include <OpenGL/gl.h>	/* Header File For The OpenGL Library */
 #include <OpenGL/glu.h>	/* Header File For The GLU Library */
@@ -12,8 +11,5 @@
 #include <GL/gl.h>	/* Header File For The OpenGL Library */
 #include <GL/glu.h>	/* Header File For The GLU Library */
 #endif
-#undef __glext_h_
-
-#include "../glext.h"
 
 #endif
